resources/aws/launch_configuration.go

```diff
@@ -0,0 +1,75 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	microerror "github.com/giantswarm/microkit/error"
+)
+
+// LaunchConfiguration is a template for launching EC2 instances into an auto
+// scaling group.
+type LaunchConfiguration struct {
+	Name                   string
+	IamInstanceProfileName string
+	ImageID                string
+	InstanceType           string
+	KeyName                string
+	SecurityGroupID        string
+	SmallCloudConfig       string
+
+	// Dependencies
+	Client *autoscaling.AutoScaling
+}
+
+// CreateIfNotExists creates the launch config if it doesn't exist.
+func (lc *LaunchConfiguration) CreateIfNotExists() (bool, error) {
+	if lc.Client == nil {
+		return false, microerror.MaskAny(clientNotInitializedError)
+	}
+
+	if err := lc.CreateOrFail(); err != nil {
```

Comment:
We do it in a few different ways unfortunately, but this seems to me like the best approach: https://github.com/giantswarm/aws-operator/blob/master/resources/aws/subnet.go#L62